addons/LPCAnimatedSprite/LPCAnimatedSprite2D.gd

```diff
@@ -34,13 +34,50 @@ enum LPCAnimation {
 	IDLE_RIGHT,
 	HURT_DOWN_LAST
 }
+
+var cur_dir:Vector2
 var AnimationNames:Array
+var reference_sheet := LPCSpriteSheet.new().NormalAnimationData
+
 func _ready():
 	if Engine.is_editor_hint() == false:
 		LoadAnimations()
-		
-func play(animation: LPCAnimation):
+		cur_dir = LookupDir(LPCAnimation.keys()[DefaultAnimation])
+
+func LookupDir(anim_name: String) -> Vector2:
+	for anim_data in reference_sheet:
+		if anim_data.Name == anim_name:
+			return anim_data.Direction
+	push_error("Supplied animation name: " + anim_name + " was not found.")
+	return Vector2.ZERO
+
+func LookUpAnimation(anim_type_name: String, direction: Vector2, oversize: bool) -> LPCAnimation:
```

Comment:
I'm not happy with this function being called everytime you call play, it iterates through the animations and after it uses a find(). My suggestion is to create a dictionary like this for all animations:

```GDScript
var AnimationDict := {
	"WALK" : {
		Vector2.UP : LPCAnimation.WALK_UP,
		Vector2.DOWN: LPCAnimation.WALK_DOWN,
		Vector2.LEFT : LPCAnimation.WALK_LEFT,
		Vector2.RIGHT: LPCAnimation.WALK_RIGHT,
	}
}
```

and you can just use it as:

```GDScript
func LookUpAnimation(anim_type_name: String, direction: Vector2, oversize: bool) -> LPCAnimation:
	return AnimationDict[anim_type_name][direction]
```

Reply:
I had this approach in the previous commit, https://github.com/alextrevisan/LPCAnimatedSprite2D/pull/22/commits/2289a53d8ebc8484ba380debb1ba60281d85c10e , your point is good, it is computationally lighter by a good margin.

Reply:
Yes, and if you map it back to the native animation type (like LPCAnimation.WALK_UP), this will work as expected when I re-do the animation with precompiled spritesheets

Reply:
Just think about defining a static type like an enum for the generic animation  too. It's better because you do not need to worry about the spelling when you call play("walk") and it does not work because it expect "WALK"

Reply:
> Just think about defining a static type like an enum for the generic animation too. It's better because you do not need to worry about the spelling when you call play("walk") and it does not work because it expect "WALK"

Makes sense. Just wondering for simplicity's sake, then, would you be ok replacing the current enum with a simplified enum + directions (for default, etc.)?

Reply:
> Yes, and if you map it back to the native animation type (like LPCAnimation.WALK_UP), this will work as expected when I re-do the animation with precompiled spritesheets

Can I safely assume there will be a separate enum entry for precompiled oversized animations?

Reply:
For now, just for sake of testing, this is how I'm doing : https://github.com/alextrevisan/LPCAnimatedSprite2D/pull/21/files#diff-38ad683d009d42b4b58f9b9dd96106b18605a7bab6480a2188756032de1b22e7R115

The animation names are the same, but when you have a spritesheet with, in this case that I have "SLASH" it replaces the normal slash with the oversized configured on the `PreSlash`

PS.: the names of these variables are temporary for now

Reply:
I don't think that we will need an oversized animation name, none of the examples that I saw had slash or other animation that are coexistent

Reply:
Sounds great, I really like that approach. The only exception would be the 2nd version of slash in the 6-6-8 pattern, if memory serves.